### PR TITLE
fix: update backend-new circleci job

### PR DESCRIPTION
**Pull request by ludtkemorgan**
Fixes the broken circle ci jobs for the backend_new jest tests and linting.

A couple of call outs:
- Had to update the e2e test to clear all of the listing db tables before each run instead of doing a reseed as the `psql` command doesn't exist
- The top level linting task now does linting of all packages besides backend_new and then does backend_new separately
- Fixed several eslint errors that have existed for a long time. There are still 40ish warnings but no errors

### CODE REVIEW

File: backend_new/test/integration/listing.e2e-spec.ts [hunk 1]
```diff
@@ -23,20 +23,19 @@ describe('Listing Controller Tests', () => {
     app = moduleFixture.createNestApplication();
     prisma = moduleFixture.get<PrismaService>(PrismaService);
     await app.init();
+    await clearAllDb();
```
Comment (YazeedLoonat) [inline backend_new/test/integration/listing.e2e-spec.ts:26]: really like this change!

File: backend_new/src/services/listing.service.ts [hunk 2]
```diff
@@ -117,7 +117,16 @@ export class ListingService {
     this set can either be paginated or not depending on the params
     it will return both the set of listings, and some meta information to help with pagination
   */
-  async list(params: ListingsQueryParams) {
+  async list(params: ListingsQueryParams): Promise<{
+    items: ListingGet[];
```
Comment (YazeedLoonat) [inline backend_new/src/services/listing.service.ts:121]: is adding the type explicitly here a required change or do you feel like this is a better pattern (making the return type as explicit as possible I mean)

Comment (ludtkemorgan) [reply backend_new/src/services/listing.service.ts:121]: This change shouldn't theoretically be needed as the type can be inferred by the returned object. However, when running the tests in ci it was saying it was "unknown" type. There is potentially a configuration change that would make it better pick up types

**Review by emilyjablonski — APPROVED** [review]
🥮

**Review by YazeedLoonat — APPROVED** [review]
LGTM

**Pull request merged**